frameworks/helloworld/tests/test_sidecar.py

```diff
@@ -31,7 +31,32 @@ def configure_package(configure_security):
 
 @pytest.mark.sanity
 def test_envvar_accross_restarts():
+
+    class ConfigException(Exception):
+        pass
+
+    def assert_envvar_has_value(envvar: str, expected_value: str):
+        _, stdout, _ = sdk_cmd.service_task_exec(config.SERVICE_NAME, "hello-0-server", "env")
+        env = dict(l.strip().split("=", 1) for l in stdout.strip().split('\n'))
+        val = env.get(envvar, "absent")
+
+        if val == "absent":
+            raise ConfigException("Required envvar not found")
+
+        if val != expected_value:
+            log.error("Looking for %s=%d but found: %s", envvar, sleep_duration, val)
```

Comment:
surprised `sleep_duration` doesn't err with `value referenced before assignment`
but anyway shouldn't `sleep_duration` be `expected_value`?

Reply:
Good catch! Python won't catch this until the line is actually executed.